plugins/nf-wave/src/main/io/seqera/wave/plugin/WaveClient.groovy

```diff
@@ -79,7 +79,7 @@ class WaveClient {
                         'Accept','application/vnd.docker.distribution.manifest.v2+json',
                         'Accept','application/vnd.docker.distribution.manifest.list.v2+json' }
 
-    private static final List<String> DEFAULT_CONDA_CHANNELS = ['conda-forge','defaults']
+    private static final List<String> DEFAULT_CONDA_CHANNELS = ['conda-forge','bioconda','defaults']
```

Comment:
Let's use [this](https://github.com/nextflow-io/rnaseq-nf/blob/master/nextflow.config#L41) (order matter)

Reply:
Order does indeed matter, and `conda-forge` must have higher priority than `bioconda`. https://bioconda.github.io/#explain-commands

```suggestion
    private static final List<String> DEFAULT_CONDA_CHANNELS = ['seqera','conda-forge','bioconda','defaults']
```

Reply:
Then pulling @jordeu about conda channels, I remember this was discussed and, if I'm not wrong, it was said to use `seqera,bioconda,conda-forge,defaults`

Reply:
The logic one would be `seqera,bioconda,conda-forge,defaults`. And I think this was the recommended some time ago by bioconda. I see that now they say "note that the recommended configuration has changed over the years" and they recommend to give `conda-forge` more priority (even I do not understand why... unless there is some versioning mess on bioconda channel) 

So, @ewels suggestion seems the correct one.

Reply:
but then it should be `conda-forge,bioconda,seqera,defaults

Reply:
Ok, I see that bioconda now also recommends to set `channel_priority: strict` that is not the default behavior (it's `flexible` the default). This may be the reason to set conda-forge before bioconda.
https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html

Not sure that I understand how `flexible` and `strict` work, specially when setting a different architecture

Reply:
what a mess :D

Reply:
I remember this going around a few years ago. I think that there are some deps on bioconda that are outdated but kept for reproducibility purposes, but conda-forge should be the preferred option for maintaining non-bioinformatics dependencies. So that's why it gets the higher priority. Or something like that, anyway.

It was quite a while ago now. I had lots of people complaining that running `conda install -c bioconda multiqc` didn't work any more...

Reply:
ok, when you feel ready mark the PR as "Ready for review"

Reply:
My questions:

1. Does top priority go first in the string (left) or last (right)?
2. Should the `seqera` channel have top priority? Or before / after bioconda?
3. Do we need to set it to strict mode somewhere, if so how. And will it break stuff @jordeu ? 😬

Reply:
Good questions :smile: 

1. left = top priority

The other two I've no idea. :cowboy_hat_face:

Reply:
Poking around for where to add conda settings to look where to set `strict`, I get back to this line:

https://github.com/nextflow-io/nextflow/blob/0bce2be098bfd61bd6d78eb5e5379fd1002fea15/plugins/nf-wave/src/main/io/seqera/wave/plugin/config/WaveConfig.groovy#L23

So are the default conda options set in the Wave codebase?

Reply:
There's no control over `strict` in nextflow and wave

Reply:
Can support for this be added somehow? From the [conda docs](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#strict-channel-priority):

> Strict channel priority can dramatically speed up conda operations and also reduce package incompatibility problems. We recommend setting channel priority to "strict" when possible.

Reply:
Any clue how this is set in the Conda yaml spec?

Reply:
After poking around a bit, I don't think that it is possible to set this in conda env YAML files 😞 

However, it can be set in a `.condarc` config file:

```yaml
channel_priority: strict
```

These files can also contain default channel priorities too, see [here](https://conda-forge.org/docs/user/tipsandtricks.html#how-to-fix-it). So if we use a `.condarc` file for global settings then we wouldn't have to specify channels in every env file.

According to [this SO comment](https://stackoverflow.com/questions/58555389/set-the-channel-priority-in-conda-environment-yaml#comment117120468_65983247), `strict` mode will become the default in Conda 5? But conda has since changed to calendar versioning, and I haven't found any other mention of it 😓

Reply:
Umm, then should it be added to the CLI? This what is run to build the image 

https://github.com/seqeralabs/libseqera/blob/84f6a27de06901da7d1101c63ded647a5d291dc3/wave-utils/src/main/resources/templates/conda/dockerfile-conda-file.txt#L1-L7

Reply:
Possibly, I'm getting a bit lost with all the different places that parts of this are happening. Wherever the conda environment is created (both Nextflow when using conda, and Wave when building from conda?), the `.condarc` file would go in the root directory of that environment. I think.

[docs](https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#channel-locations-channels):
> To select channels for a single environment, put a `.condarc` file in the root directory of that environment (or use the --env option when using `conda config`).

Reply:
Alternatively, I wonder if this is getting too complicated. Maybe we just set the channels as in this PR and leave it to the Nextflow user to configure conda appropriately? That's already an improvement over the current status, and it's possible that setting this config won't be a requirement for very much longer anyway.

As I read it, having the default `channel_priority` will mostly make the environment resolution slower, not sure that it will necessarily break (unless you disagree @jordeu?).

Could then just drop a `.condarc` in the home directory of the wave base image for conda that says `channel_priority: strict` and we should be good to go...

Reply:
Reading the docs I believe if we use this, the seqera channel is not going to work anymore 

> Setting strict channel priority makes it so that if a package exists on a channel, conda will ignore all packages with the same name on lower priority channels.

Reply:
I don't know

Reply:
For me from the documentation is not clear what happens with the architecture restriction, if they apply before or after building the list of candidates. 

I'd say the `seqera,conda-forge,bioconda,defaults` and `flexible` (the default) channel priority should work well in "all" cases.

Reply:
ok, let's merge this